### ISSUE TITLE
Slicer Tutorial Maker : Test 12/12/2024

### ISSUE THREAD

**Issue by MohamedAllali** [SlicerLatinAmerica/SlicerTutorialMaker#55]
@spujol @vmms @LuskasHusty @DouSam @FatouBintou22
These are the issues we encountered when we tested the new version of the TutorialMaker. 

**For Slicer 5.6.2**

![image](https://github.com/user-attachments/assets/d471b6ee-a7ad-44fc-acc1-0d3c0ef14310)

**a) fourMin_tutorial** 
**Issues**

- [x] screenshots that are not part of the tutorial still shows up in the new version

![image](https://github.com/user-attachments/assets/b6bc3471-dcb0-48a4-ae64-e3688124bf96)

- [ ] The screenshots do not appear correctly in the generated pdf version. There are empty pages that are not part of the tutorial and appear in the pdf.

![image](https://github.com/user-attachments/assets/ae02bd35-bb9f-497f-ad9d-3ebe9cd451eb)

- [ ] The “delete” button and arrow does not work. See issue [#42](https://github.com/SlicerLatinAmerica/SlicerTutorialMaker/issues/42) 

![image](https://github.com/user-attachments/assets/c081546c-3c90-4292-b42b-1723fa3efc39)

- [x] The choice here should be white matter and not skull_bone as it exists in original tutorial

![image](https://github.com/user-attachments/assets/431cdcd1-01d9-43b1-af6d-f5d36cab11af)

This is the original tutorial:

![image](https://github.com/user-attachments/assets/404246d0-b14a-4755-84ef-fda3eb9be1d3)

**For Slicer-5.7.0-2024-12-10
Tutorial Maker via extension manager**
 
![image](https://github.com/user-attachments/assets/c2511277-35dc-487b-819f-71a977d43b69)

**a) welcome_tutorial, fourMin_tutorialc, welcome_tutorialc**

- [x] fourMin_tutorialc, welcome_tutorialc ne s’execute pas. 

- [x] welcome_tutorial is executed but the Annotator does not display.

**b) fourMin_tutorial
Issues**
- [ ] There is no slide Acknowledgement 
- [ ] The PDF annotator button is enabled but the pdf file is not generated 
- [ ] The same problems encountered with the Slicer version 5.6.2 are present

 
![image](https://github.com/user-attachments/assets/d5f3131d-e2f6-4f2f-a1c4-d4dd393e6778)

**Tutorial Maker cloned via GitHub**
 
![image](https://github.com/user-attachments/assets/24e83030-612f-4fc6-a569-4afb04d3e43c)

**a) welcome_tutorial** 

- [ ] welcome_tutorial is executed but the **Annotator** does not display

 
![image](https://github.com/user-attachments/assets/8ded2568-40cf-4d07-8137-f3e8791c0d46)

**b) fourMin_tutorial
Issues**

- [ ] The same problems encountered with the Slicer version 5.6.2 are present

![image](https://github.com/user-attachments/assets/dba94cde-6879-444d-bc78-e4ba88a17a7c)
 
 
![image](https://github.com/user-attachments/assets/73a52e2c-9595-478b-8786-925706357432)

![image](https://github.com/user-attachments/assets/fdc81cae-2f3a-4ed3-87bf-ae8d84f844ee)

**Comment by lassoan** [comment]
Please do all the test on the latest Slicer Preview Release. It is updated every night, so if something does not work as expected then download the latest and if that still does not work well then report the problem.

**Comment by DouSam** [comment]
In this issue we have some problems that already have issues opened;
So, first the problem 

> "screenshots that are not part of the tutorial still shows up in the new version"

That as solved in today version, you can manually update the extension or, as suggested by andras, wait for the next preview version.

> "The screenshots do not appear correctly in the generated pdf version. There are empty pages that are not part of the tutorial and appear in the pdf."
@MonseRiosH @enriquehdez98 @vmms  could you check this? Let us know if you need any help to solve

> "The “delete” button and arrow do not work. See issue https://github.com/SlicerLatinAmerica/SlicerTutorialMaker/issues/42"
Same here @enriquehdez98 and @vmms 

> "The choice here should be white matter and not skull_bone as it exists in original tutorial"
I will mark this as done. We talked some meetings ago, and we have used the Sonia tutorial version, that's the version available on Slicer's official page. Open an issue in https://github.com/SlicerLatinAmerica/TestSlicerTutorials if you need a new tutorial using your version!

For the other errors and problems, please @MohamedAllali and @FatouBintou22 manually update your extension, delete the existing version, download the last version, and see if the problems still occurring. If yes, let me know, if not, mark the tasks as done! To ensure that the version you are working on is the last, see the "Print" in the console, that has the version of the extension.

![image](https://github.com/user-attachments/assets/6ecce23d-c079-476a-9901-dd2d9b6e3449)

**Comment by MohamedAllali** [comment]
Thank you @DouSam for your feedback. 
I downloaded the new version of Tutorial Maker and tested it on the latest version of slicer, and it seems that the PDF feature no longer works. When I click on the PDF icon, nothing happens.

**Comment by LuskasHusty** [comment]
@MohamedAllali If I'm not mistaken we are currently changing how the pdf is created. It should be faster and not freeze slicer for as long, and also not freeze on first execution.

**Comment by LuskasHusty** [comment]
Yep no problem, the new version as of right now should have fixed a lot of the screenshot related programs, if you could give feedback on that today it would be very helpful.
Also note that the new pdf creation should be working very shortly, hopefully before today's meeting.

**Comment by MohamedAllali** [comment]
> Yep no problem, the new version as of right now should have fixed a lot of the screenshot related programs, if you could give feedback on that today it would be very helpful. Also note that the new pdf creation should be working very shortly, hopefully before today's meeting.

True, I noticed that screenshots became better, especially after fixing the appearance of screenshots that do not belong to tutorials.

**Comment by DouSam** [comment]
Since we refactored all the code and created a new version, the errors are solved now. I will open the issues for the new errors that appeared in the new version.
